src/site/_collections/authors.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const fs = require('fs');
+const path = require('path');
+const contributors = require('../_data/contributors');
+const livePosts = require('../_filters/live-posts');
+const setdefault = require('../_utils/setdefault');
+
+/**
+ * Generate map the posts by author's username/key
+ *
+ * @param {*} posts
+ * @return {Map<string, Array<any>>} Map of posts by author's username/key
+ */
+const findAuthorsPosts = (posts) => {
+  const authorsMap = new Map();
+  posts.forEach((post) => {
+    const authors = post.data.authors || [];
+    authors.forEach((author) => {
+      const postsByAuthor = setdefault(authorsMap, author, []);
+      postsByAuthor.push(post);
+      authorsMap.set(author, postsByAuthor);
```

Comment:
I thought the point of setdefault is that you don't need to set it again :)

Reply:
So this is now a oneliner:

```js
setdefault(authorsMap, author, []).push(post);
```

Reply:
Ok, so I would have liked to do this, but 11ty isn't liking it 😤:

<img width="446" alt="Screen Shot 2020-04-17 at 11 44 00 AM" src="https://user-images.githubusercontent.com/11811422/79603259-d54df480-80a0-11ea-9991-58169253fead.png">

Reply:
Just realized `push` wouldn't return the array, so we're assigning a number to `postsByAuthor`